core/src/main/java/org/apache/iceberg/BaseMetadataTable.java

```diff
@@ -73,9 +73,12 @@ protected BaseMetadataTable(Table table, String name) {
    */
   static PartitionSpec transformSpec(Schema metadataTableSchema, PartitionSpec spec) {
     PartitionSpec.Builder identitySpecBuilder =
-        PartitionSpec.builderFor(metadataTableSchema).checkConflicts(false);
+        PartitionSpec.builderFor(metadataTableSchema)
+            .withSpecId(spec.specId())
+            .checkConflicts(false);
     for (PartitionField field : spec.fields()) {
```

Comment:
nit: What about adding empty lines before and after this loop? Also, I'd consider renaming the builder variable to make it shorter and stay on one line.

Reply:
Done